tests/Spec.hs

```diff
@@ -565,6 +621,55 @@ main = hspec $ do
         , ALeaveComment (PullRequestId 1) "Rebased as 38e, waiting for CI \x2026"
         ]
 
+    it "can handle a rebase failure after a failed push" $ do
+      -- This is a regression test for the following scenario:
+      -- - A pull request goes through the approve-rebase-build cycle.
+      -- - Pushing to master fails.
+      -- - After restarting the cycle, rebasing fails.
+      -- In the past this would trigger an inconsistent state because we forgot
+      -- to clear the build status.
+      let
+        state
+          = Project.insertPullRequest
+              (PullRequestId 1)
+              (Branch "n7")
+              (Sha "a39")
+              "Add Nexus 7 experiment"
+              (Username "tyrell")
+          $ Project.emptyProjectState
+        events =
+          [ CommentAdded (PullRequestId 1) "deckard" "@bot merge"
+          , BuildStatusChanged (Sha "b71") Project.BuildPending
+          , BuildStatusChanged (Sha "b71") Project.BuildSucceeded
+          ]
+        -- For this test, the first integration succeeds. Then we push, which
+        -- fails. Then we try to integrate again, but that fails.
+        results = defaultResults
+          { resultIntegrate =
+            [ Right $ Sha "b71"
+            , Left $ Logic.IntegrationFailure (Branch "master")
+            ]
+          , resultPush =
+            [ PushRejected
+              -- TODO(ruuda): Why is this second result needed? We should not do
+              -- a second push.
+            , PushOk
```

Comment:
Will this be fixed together with the inconsistent state bug?

Reply:
Yes, once it no longer hits the error case there, I hope to be able to further debug this test.

Reply:
This has also been addressed in #59.